auslib/db.py

```diff
@@ -895,10 +896,15 @@ def _splitColumns(self, columns):
         return base_columns, condition_columns
 
     def _checkBaseTablePermissions(self, base_table_where, new_row, changed_by, transaction):
-        if new_row.get("data_version"):
+        if "change_type" not in new_row:
+            raise ValueError("change_type needed to check Permission")
+
+        if new_row.get("change_type") == "update":
             self.baseTable.update(base_table_where, new_row, changed_by, new_row["data_version"], transaction=transaction, dryrun=True)
-        else:
+        elif new_row.get("change_type") == "insert":
             self.baseTable.insert(changed_by, transaction=transaction, dryrun=True, **new_row)
+        elif new_row.get("change_type") == "delete":
+            self.baseTable.delete(base_table_where, changed_by, new_row["data_version"], transaction=transaction, dryrun=True)
```

Comment:
Best to be on the safe side and raise a ValueError in an else block here. If somehow we managed to get a Scheduled Change with an unknown change_type, we definitely don't want it to get into the database.